agent/agents/supervisor/supervisor.go

```diff
@@ -384,7 +384,11 @@ func (s *Supervisor) setBuiltinAgents(builtinAgents map[string]*agentpb.SetState
 
 // filter extracts IDs of the Agents that should be started, restarted with new parameters, or stopped,
 // and filters out IDs of the Agents that should not be changed.
-func filter(existing, ap map[string]agentpb.AgentParams) (toStart, toRestart, toStop []string) {
+func filter(existing, ap map[string]agentpb.AgentParams) ([]string, []string, []string) {
+	var toStart []string
+	toRestart := make([]string, 0, len(ap))
+	toStop := make([]string, 0, len(existing))
```

Comment:
why does `toStart` is empty array when others are pre-allocated?

Reply:
Ah, right. Changed.